docs/developer-docs/gas-cost.md

```diff
@@ -1,100 +1,212 @@
 
-Cost per Transaction in USD (XDR/USD exchange rate as of November 23, 2022).
+Each type of canister operation has a different cycles cost associated with it, and the canister responsible for paying the cycles varies based on the type of operation.
 
-The following table shows the calculated storage cost per GB for a 30-day month: 
+### Canister creation
 
-|                      |                                    | 13-node Application Subnets | 34-node Application Subnets |
-|----------------------|------------------------------------|-----------------------------|-----------------------------|
-| GB Storage Per Month | For storing a GB of data per month | $0.431                      | $1.127                      |
+Canister creation costs 100B cycles or approximately $0.133661 USD. Canisters can be created by users or other canisters.
+
+### Messaging
+
+A canister can receive messages from users and other canisters. In canister-to-canister messages, the sending canister pays message transmission costs. In user-to-canister messages, the receiving canister covers the message transmission costs (see [reverse gas model](#the-icp-reverse-gas-model). User-to-canister messages are also referred to as ingress messages.
+
+The message transmission cost consists of a fixed baseline fee and per-byte-fee charged for each byte of the message: `base-fee` + `per-byte-fee` * `size-in-bytes`.
+
+The current fees are:
+
+| Message type | Base fee | Per byte fee |
+|--------------|----------|--------------|
+| Canister-to-canister | 260K | 1K |
+| User-to-canister (ingress) | 1.2M | 2K |
+
+
+An additional factor of cost includes the number of nodes in the subnet the canister is deployed on.
+
+Note that query messages are currently free, but this may change in the future.
+
+### Execution
+
+In order to handle an incoming message, the canister executes the function specified in the message. The execution cost consists of a fixed execution fee and per-instruction fee that is charged for each executed WebAssembly instruction: `base-fee` + `per-instruction-fee` * `number-of-instructions`. The current values of fees are `base-fee` = 590K cycles (or $0.0000007885999 USD), `per-instruction-fee` = 0.4 cycles (or $0.000534644 USD for 1B instructions).
+
+By default canisters are scheduled for execution in a "best-effort" manner. Canisters that require guaranteed execution can a share of compute capacity by setting `compute_allocation` in their canister settings. Compute allocation is expressed in percents and denote the percentage of an execution core reserved for the canister. For example, compute allocation of 50% means that the canister will get 50% of an execution core. In other words, it will be scheduled at least every other round. Compute allocation of 100% means that the canister will be scheduled every round. The current fee for 1 percent computer allocation per second is 10M cycles (or $0.0000133661 USD).
```

Comment:
A word is missing: "Canisters that require guaranteed execution can _*get*_ a share of compute capacity"

Reply:
Added 'get'